client/state/utils.js

```diff
@@ -116,3 +116,54 @@ export function createReducer( initialState = null, customHandlers = {}, schema
 		return state;
 	};
 }
+
+/**
+ * Extends createReducer, returning a reducer function. The schema and return
+ * value of the enhanced reducer is augmented to create a nested object keyed
+ * by an assumed siteId property on each action handled by customHandlers.
+ *
+ * @see createReducer
+ *
+ * @param  {*}        initialSiteState Initial state per site ID
+ * @param  {Object}   customHandlers   Object mapping action types to state
+ *                                     action handlers
+ * @param  {?Object}  schema           JSON schema object for deserialization
+ *                                     validation
+ * @return {Function}                  Reducer function
+ */
+export function createSiteReducer( initialSiteState = null, customHandlers = {}, schema = null ) {
```

Comment:
Yeah, and more generally we need better documentation of all of our state utilities.